### DISCUSSION TITLE
Device does not support this mini-program (GTR3 PRO LTD)

### DISCUSSION THREAD

**Discussion by rgaida** [zepp-health/zeppos-forum#376] [General]
I’m trying to build the ZeppOS hello-world sample app for my Amazfit GTR 3 Pro Limited Edition (using `zeus create hello-world`, OS API 1.0, target: gtr3-pro). However, I’m encountering issues when attempting to test the app directly on the smartwatch.

When I try to use `zeus bridge`, I get the following error: "Install lite app result: device does not support this mini-program/watchface". Additionally, scanning the QR code generated via `zeus preview` in the Zepp app results in a "Download files, error code: null" message.

I'm running out of ideas on how to get a basic sample app running on my watch. Any advice or solutions would be greatly appreciated.

**Comment by MyoungXUE** [comment]
Hello, I reproduced your situation. The GTR 3 Pro Limited Edition device is quite special, and our use case did not support this situation at the beginning

You need to manually modify it in `app.json`.

https://docs.zepp.com/docs/reference/related-resources/device-list/

Query all deviceSources of GTR3 Pro in the document and fill them into `app.json`.

<img width="789" alt="image" src="https://github.com/user-attachments/assets/acc4c9d1-bc7b-4302-8c34-19523391709d">

```json
"platforms": [
        {
          "name": "gtr3-pro",
          "deviceSource": 229
        },
        {
          "name": "gtr3-pro-w",
          "deviceSource": 230
        },
        {
          "name": "gtr3-pro-w1",
          "deviceSource": 242
        },
        {
          "name": "gtr3-pro-wl",
          "deviceSource": 6095106
        },
      ],
      "designWidth": 480
```

**Reply by rgaida** [reply]
Thank you so much! It works as intended now. I suspected something like that, but I couldn’t find any specific documentation hinting at special treatment for the GTR3 Pro Limited Edition. Also, I wasn’t able to find a way to retrieve the device ID directly from the watch.
